common/environment/environment.cpp

```diff
@@ -1157,11 +1304,66 @@ IConstDropZoneInfo * CLocalEnvironment::getDropZoneByComputer(const char * compu
     if (!computer)
         return NULL;
     buildDropZoneCache();
-    VStringBuffer xpath("Software/DropZone[@computer=\"%s\"]", computer);
+
+    StringBuffer x("@");
+    x.append(computer).append(NUMBER_OF_DROPZONES_SUFFIX);
+    unsigned numOfDropZoneByComputer = numOfDropzonesByComputer->getPropInt(x, -1);
+    if ( numOfDropZoneByComputer == -1 )
+        return NULL;
+
+    StringBuffer xpath("Software/DropZone[@computer=\"");
+    xpath.append(computer);
+    xpath.append(DROPZONE_BY_MACHINE_SUFFIX).append(DEFAULT_DROPZONE_INDEX);
+    xpath.append("\"]");
+
     synchronized procedure(safeCache);
     return (CConstDropZoneInfo *) getCache(xpath.str());
 }
 
+IConstDropZoneInfo * CLocalEnvironment::getDropZoneByComputer(const char * computer, const char * dzname) const
+{
+    IConstDropZoneInfo * dzInfo = getDropZone(dzname);
+    if (!dzInfo)
+        return NULL;
+
+    SCMStringBuffer cachedComputer;
+    dzInfo->getComputerName(cachedComputer);
+
+    if ( strcmp(computer, cachedComputer.str()) != 0)
+        return NULL;
+
+    return dzInfo;
+
+}
+
+IConstDropZoneInfo * CLocalEnvironment::getDropZoneByComputerByIndex(const char * computer, unsigned index) const
+{
+    if (!computer)
+        return NULL;
+    if ((index == 0 ) || (index > getNumberOfDropZonesByComputer(computer)) )
+        return NULL;
+
+    buildDropZoneCache();
+
+    StringBuffer xpath("Software/DropZone[@computer=\"");
+    xpath.append(computer).append(DROPZONE_BY_MACHINE_SUFFIX).append(index).append("\"]");
+    synchronized procedure(safeCache);
+    return (CConstDropZoneInfo *) getCache(xpath.str());
+}
```

Comment:
The iterators and direct get by name are enough, getting by computer+index as well pollutes the interface.
Better to remove.